File: dart-language/charge-system-demo/client-ng-dart/lib/app_component.dart
import 'package:angular/angular.dart';
import 'package:angular_router/angular_router.dart';

import 'src/routes.dart';
import 'src/user_service.dart';
import 'src/charge_service.dart';
import 'src/uri_service.dart';


@Component(
  selector: 'my-app',
  styleUrls: ['app_component.css'],
  templateUrl: 'app_component.html',
  providers: [ClassProvider(UserService), ClassProvider(ChargeService), ClassProvider(UriService)],
  directives: [routerDirectives],
  exports: [RoutePaths, Routes]
)
class AppComponent {
}

File: angular-playlist/first-app/src/app/app.component.ts
import {Component, OnInit} from '@angular/core';
import {Router} from "@angular/router";
import {APP_ROUTES} from "./router/app-routing.module";

@Component({
    selector: 'app-root',
    templateUrl: './app.component.html',
    styleUrls: ['./app.component.css']

})
export class AppComponent implements OnInit {
    title = 'Tour of Heroes';
    navs = null;

    constructor(private router: Router) {
    }

    ngOnInit() {
        this.navs = APP_ROUTES.filter(v => v.path !== '').map(v => ['/' + v.path]);
    }

    onNav() {
        this.router.navigate(['heroes']);
    }
}
